### PR TITLE
updates some functionality and removes some permissions we don't want

### PR DESCRIPTION
**Pull request by pixelknitter**
A brief summary of changes:
- We found various issues with how CallKeep returned information for the inputs and outputs, which are fixed here.
- Additionally, we've temporarily removed unnecessary permissions for our use case on Android.

### CODE REVIEW

File: ios/RNCallKeep/RNCallKeep.m [hunk 1]
```diff
@@ -188,11 +196,15 @@ + (void)initCallKitProvider {
     }
 }
 
-+ (NSString *) getAudioOutput {
++ (NSMutableDictionary *) getAudioOutput {
```
Comment (dfinn) [inline ios/RNCallKeep/RNCallKeep.m:199]: I see the contents here differ slightly from our patch -- should I assume you created the fork against the latest main branch on the upstream repo, and there must be some new changes there that we didn't pick up yet?  
The differences look like they should be fine, but let's make sure we re-test all this stuff...

**Review by dfinn — APPROVED** [review]
The changes look fine, although I have not tested them.

File: ios/RNCallKeep/RNCallKeep.m (hunk 1 above)
Comment (pixelknitter) [reply ios/RNCallKeep/RNCallKeep.m:199]: Correct, the main thing they added was a try/catch here:
image

Comment (pixelknitter) [reply ios/RNCallKeep/RNCallKeep.m:199]: This was to capture and resolve a crash that was occurring here.

**Pull request merged**